lib/tests/test_gpg.rs

```diff
@@ -143,8 +143,9 @@ fn gpg_signing_roundtrip_explicit_key() {
         display: Some(
             "Someone Else (jj test signing key) <someone-else@example.com>",
         ),
+        backend: None,
```

Comment:
I suppose it's alright that `backend` is `None` here, since the `Signer` is setting it and not the backend?

Reply:
I did not question if there is a reason that (or if at all) the backend should always be set by the `Signer`.
@necauqua, do you know?

Reply:
Nope
As I said, the only recollection I have of the backend field at all is that it could be used in the templater?.
Like the `builtin_log_detailed` would have another line looking like `Signature: A good **GPG** signature by Test User <user@example.com>` etc.

So if it is not obvious why the backend field is even needed it can be ditched (I think Yuja suggested that, we can think if it's really needed later), the whole "setting the field from the outside" seems dodgy anyway

The answer to the direct question "why the backend shouldn't set its own name" I think was something along the lines of "reducing things backend impl should care about" - so nothing special.

Reply:
So if we prefer strictness, the `Verification` type would be `Unknown|Verified(BackendVerification, BackendName)`. I don't have a strong feeling about that, so I suggested either making `backend` public or removing it for now.

> I think Yuja suggested that, we can think if it's really needed later

Yes.

Reply:
All right, thanks again for clarifying. I didn't really get it the first time.
I will remove the `backend` field then and we can see how we deal want to deal with it later.

Reply:
> So if we prefer strictness, the Verification type would be Unknown|Verified(BackendVerification, BackendName).

That pseudo code made it click for me.

Now, with `backend` removed, the diff is also much more focused and concise. 👍